### ISSUE TITLE
After remove game, proton prefix exist

### ISSUE THREAD

**Issue by bladefox1337** [ValveSoftware/steam-for-linux#7418]
#### Your system information

* Steam client version (build number or date): Oct 15 2020, at 19:03:06
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
After remove windows game from library, proton prefix remains in ~/.local/share/Steam/steamapps/compatdata/%gameid%

#### Steps for reproducing this issue:

1. Install & play game via proton (example Brawlhalla[291550])
2. Remove game
3. Check %gameid% for your game, and open ~/.local/share/Steam/steamapps/compatdata directory, and %gameid% directory is exist

**Comment by kisak-valve** [comment]
Possibly intentional to mitigate https://github.com/ValveSoftware/Proton/issues/428.

**Comment by kisak-valve** [comment]
[Feature Request] Feature to clear proton compatdata for uninstalled apps

Issue transferred from https://github.com/ValveSoftware/steam-for-linux/issues/7815.
@xDShot posted on 2021-05-26T09:49:15:

Was cleaning disk space when I noticed that I had leftover compat datas from removed games in steam/steamapps/compatdata. Medium size of these is around 500Mb, some of them even took 20Gbs, and there are plenty of them.

Would like to see an option for Steam client to purge chosen data of willing app, including for removed apps, instead of manually removing them. Optional as they might contain save data for games without Steam Cloud or have some prefix workaround applied to make them work.

**Comment by techmouse** [comment]
I'm kind of embarrassed to post this but I made a BASH script quite some time ago for my own personal usage. It was originally designed to solve a slightly different problem, but I've been adding to it over time and it has come to address this exact issue.

https://github.com/techmouse/steam-compatdata-symlinker-thing

Let me know if you get some use out of it.

**Comment by IBNobody** [comment]
> I'm kind of embarrassed to post this but I made a BASH script quite some time ago for my own personal usage. It was originally designed to solve a slightly different problem, but I've been adding to it over time and it has come to address this exact issue.
> 
> https://github.com/techmouse/steam-compatdata-symlinker-thing
> 
> Let me know if you get some use out of it.

It's very good, but it has some issues on the Steam Deck. I'll file an issue against it.

**Comment by MagicD3VIL** [comment]
**[Feature Request]** While uninstalling a SteamPlay/Proton game, add a checkbox asking "Do you want to delete all user data?". If the user does not check this box, do **not** delete the prefix. If checked, delete the prefix.

**Comment by 62832** [comment]
Would be nice to have this added as a native option to Steam, though for the time being I made myself a relatively simple Bash script [here](https://github.com/62832/pfxs) that I've been using to clear away prefixes for uninstalled games. Maybe others here might find some use in it as well.

**Comment by 62832** [comment]
Has this been fixed now in the new Beta client update? Looks like unused prefixes no longer actually remain when uninstalling a game.

**Comment by Saroumane** [comment]
Erasing player saves when uninstalling games ? That takes us back to DOS gaming era !
Remember, Valve : not all games have cloud saves (and even with cloud saves, data loss can happen)
Seriously : it's never been recommended (euphemism) to erase user data without consent when uninstalling a program. 

@MagicD3VIL  is so right here : https://github.com/ValveSoftware/steam-for-linux/issues/7418#issuecomment-1247317571  it must be a **choice** with a big warning. And it must warn about the saves.
=> "Do you want to delete all user data **(including local saves)** ?"

**Comment by ASHGOLDOFFICIAL** [comment]
Today I've lost my game progress (about two weeks) when I reinstalled the game (otherwise it wouldn't open) because Steam also deleted my local saves and didn't create any Cloud copies. So yeah, I think there should be choice to save local data rather deleting it permanently. At least send it to trash.

**Comment by techmouse** [comment]
> At least send it to trash.

I completely agree with this sentiment. Always use `gio trash` instead of `rm`.
